yoti-sdk-spring-boot-example/src/main/java/com/yoti/api/examples/springboot/IdentityLoginController.java

```diff
@@ -70,9 +70,14 @@ public String identityShare(Model model) {
     @GetMapping(value = "/receipt")
     public String receipt(@RequestParam("receiptId") String receiptId, Model model) {
         Receipt receipt = execute(() -> client.fetchShareReceipt(receiptId), model);
+        if (receipt == null) {
+            return "error";
+        }
+
+        if (receipt.hasError()) {
+            receipt.getError().ifPresent(error -> model.addAttribute("error", error));
+            receipt.getErrorReason().ifPresent(reason -> model.addAttribute("errorReason", reason));
 
-        if (receipt == null || receipt.getError().isPresent()) {
-            model.addAttribute("error", receipt.getError().get());
             return "error";
```

Comment:
Maybe extract that

        if (receipt == null || receipt.hasError()) {
            return handleError();
        }

and 

    private static handleError() {
        if (receipt != null) {
            receipt.getError().ifPresent(error -> model.addAttribute("error", error));
            receipt.getErrorReason().ifPresent(reason -> model.addAttribute("errorReason", reason));
        }

        return "error";
    }